rts/System/Sound/OpenAL/MusicStream.cpp

```diff
@@ -176,13 +176,14 @@ bool MusicStream::StartPlaying()
 {
 	msecsPlayed = spring_nulltime;
 	lastTick = spring_gettime();
-
-	if (!DecodeStream(buffers[0]) || !DecodeStream(buffers[1])) {
-		// streaming very small file is not possible if we use 2 buffers
+	
+	if (!DecodeStream(buffers[0])) {
 		return false;
 	}
+	
+	const int numDecodedStreams = 1 + static_cast<int>(DecodeStream(buffers[1]));
```

Comment:
1) What if it's a big sound file but decoding fails?
2) I slightly dislike this cast, especially since it doesn't allow any error handling. How about something like

```cpp
int numDecodedStreams = 1;
if (DecodeStream(buffers[1])) {
  ++ numDecodedStreams;
} else {
  // did it fail, or was it just a very small file?
}
```